testing/testing/src/main/java/io/helidon/testing/TestConfig.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.testing;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantLock;
+
+import io.helidon.common.context.ContextSingleton;
+
+/**
+ * Static accessor to {@link io.helidon.testing.TestConfigSource} to be able to set values.
+ */
+public final class TestConfig {
+    private static final ContextSingleton<Sources> SOURCES = ContextSingleton.create(TestConfig.class,
+                                                                                     Sources.class,
+                                                                                     Sources::new);
+    private static final ContextSingleton<AccumulatedOptions> ACCUMULATED_OPTIONS = ContextSingleton.create(
+            TestConfig.class,
+            AccumulatedOptions.class,
+            AccumulatedOptions::new);
+    private static final ReentrantLock LOCK = new ReentrantLock();
+
+    private TestConfig() {
+    }
+
+    /**
+     * Set a configuration key and value.
+     * This method CANNOT override an existing key, as such keys are already in the config snapshot.
```

Comment:
FAQ topic candidate

Reply:
Do we event need this method ?
This looks like a hack that we aren't using, at least in this pull request.

One should be able to use this instead:
```java
@TestConfig.Source
static ConfigSource config() {
    var config = TestConfigSource.create();
    onPseudoEvent(() -> config.set("foo", "bar"));
    return config;
}
```